### ISSUE TITLE
dynamoDB Superset support for queryDB 

### ISSUE THREAD

**Issue by zawadal** [passren/PyDynamoDB#57]
Hello! Wondering if this is a bug or my misunderstanding of caching features based on QueryDB file query.db (not in-mem). I've configured ENVs according to following guidance: https://github.com/passren/PyDynamoDB/wiki/5.-Superset#other-type-of-querydb-support-and-cache-044. 

Config for my needs:

`
"PYDYNAMODB_QUERYDB_TYPE": "sqlite",
"PYDYNAMODB_QUERYDB_URL": "/app/query.db",
"PYDYNAMODB_QUERYDB_LOAD_BATCH_SIZE": "200",
"PYDYNAMODB_QUERYDB_EXPIRE_TIME": "3600",
"PYDYNAMODB_QUERYDB_PURGE_ENABLED": "True",
"PYDYNAMODB_QUERYDB_PURGE_TIME": "86400",`

When trying to execute any query against dynamoDB (previously working fine with `:memory` for PYDYNAMODB_QUERYDB_URL) i got following error:

```2024-11-25 12:53:16,568:WARNING:superset.views.error_handling:SupersetErrorsException
Traceback (most recent call last):
File "/usr/local/lib/python3.10/site-packages/flask/app.py", line 1484, in full_dispatch_request	
rv = self.dispatch_request()
File "/usr/local/lib/python3.10/site-packages/flask/app.py", line 1469, in dispatch_request
return self.ensure_sync(self.view_functions[rule.endpoint])(**view_args)
File "/usr/local/lib/python3.10/site-packages/flask_appbuilder/security/decorators.py", line 95, in wraps	
return f(self, *args, **kwargs)	
File "/app/superset/views/base_api.py", line 119, in wraps	
duration, response = time_function(f, self, *args, **kwargs)	
File "/app/superset/utils/core.py", line 1364, in time_function
response = func(*args, **kwargs)	
File "/app/superset/views/base_api.py", line 91, in wraps
return f(self, *args, **kwargs)	
File "/app/superset/utils/log.py", line 303, in wrapper	
value = f(*args, **kwargs)
File "/app/superset/sqllab/api.py", line 409, in execute_sql_query
command_result: CommandResult = command.run()	
File "/app/superset/utils/decorators.py", line 266, in wrapped	
return on_error(ex)	
File "/app/superset/utils/decorators.py", line 236, in on_error
raise ex
File "/app/superset/utils/decorators.py", line 259, in wrapped
result = func(*args, **kwargs)
File "/app/superset/commands/sql_lab/execute.py", line 105, in run
status = self._run_sql_json_exec_from_scratch()
File "/app/superset/commands/sql_lab/execute.py", line 158, in _run_sql_json_exec_from_scratch
return self._sql_json_executor.execute(
File "/app/superset/sqllab/sql_json_executer.py", line 111, in execute
raise SupersetErrorsException(

superset.exceptions.SupersetErrorsException: [SupersetError(message='dynamodb error: no such table: QUERYDB_CACHES', error_type=<SupersetErrorType.GENERIC_DB_ENGINE_ERROR: 'GENERIC_DB_ENGINE_ERROR'>, level=<ErrorLevel.ERROR: 'error'>, extra={'engine_name': 'Amazon DynamoDB', 'issue_codes': [{'code': 1002, 'message': 'Issue 1002 - The database returned an unexpected error.'}]})]
```

This error seems to indicate the DB it's trying to reach for QUERYDB_CACHES is Amazon DynamoDB (where my DATA resides) which should not be the case right? It should use sqlite DB under the hood if I'm not mistaken (It's not configured in Superset as DB). Additionally I can see in code the QUERYDB_CACHES should be automatically created and populated by pydynamodb engine so only thing I provided was empty query.db file created using sqlite3 command.

![obraz](https://github.com/user-attachments/assets/7f904093-8e0d-40a1-ab33-d3672d6f11e3)

What am I missing here? Should QueryDB be preconfigured somehow? Objective is to provide some sort of dynamoDB scan/query caching which seems to require file queryDB not :memory: right?

**Comment by zawadal** [comment]
> @zawadal QUERYDB_CACHES is an internal table which supposes to be created automatically once QueryDB getting initialized. Can you check if sqlite file "/app/query.db" is there?

Yup - I'm creating this file in Dockerfile using 'sqlite3 query.db "VACUUM;"' and applying superset:superset owner and 774 permissions - otherwise Superset complains the file do not exist . So in the moment of superset container initialization backend file is already there waiting empty

**Comment by zawadal** [comment]
Thanks for all the effort and working on this package, sir.

**Comment by passren** [comment]
@zawadal This issue is getting fixed in 0.7.1
